### PR TITLE
Bump native provider to version 5.19.0

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

**In this PR:**

- Native provider version upgraded to 5.15.0

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Changes made in this PR have been tested uptest and manually, please see comments.

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/budgets/budget.yaml"

**Comment by turkenf** [comment]
**Breaking CRD schema changes**

- `package/crds/acmpca.aws.upbound.io_certificateauthorities.yaml`
Manually:
- `package/crds/apigateway.aws.upbound.io_restapis.yaml`
Manually:
- `package/crds/autoscaling.aws.upbound.io_attachments.yaml`
Manually:
- `package/crds/autoscaling.aws.upbound.io_autoscalinggroups.yaml`
Manually:
- `package/crds/autoscaling.aws.upbound.io_launchconfigurations.yaml`
Manually:
- `package/crds/budgets.aws.upbound.io_budgets.yaml`
Uptest: https://github.com/upbound/provider-aws/actions/runs/6089878111
- `package/crds/connect.aws.upbound.io_hoursofoperations.yaml`
Uptest:
- `package/crds/connect.aws.upbound.io_queues.yaml`
Uptest:
- `package/crds/connect.aws.upbound.io_routingprofiles.yaml`
Uptest:
- `package/crds/directconnect.aws.upbound.io_connections.yaml`
Uptest:
- `package/crds/dms.aws.upbound.io_endpoints.yaml`
Manually:
- `package/crds/ec2.aws.upbound.io_defaultvpcs.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_routes.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_routetables.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_vpcpeeringconnectionaccepters.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_vpcpeeringconnectionoptions.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_vpcpeeringconnections.yaml`
Uptest:
- `package/crds/ec2.aws.upbound.io_vpcs.yaml`
Uptest:
- `package/crds/ecs.aws.upbound.io_clusters.yaml`
Uptest:
- `package/crds/elasticache.aws.upbound.io_clusters.yaml`
Uptest:
- `package/crds/elasticache.aws.upbound.io_replicationgroups.yaml`
Uptest:
- `package/crds/firehose.aws.upbound.io_deliverystreams.yaml`
Uptest:
- `package/crds/iam.aws.upbound.io_roles.yaml`
Uptest:
- `package/crds/kafka.aws.upbound.io_clusters.yaml`
Manually:
- `package/crds/lightsail.aws.upbound.io_instances.yaml`
Uptest:
- `package/crds/medialive.aws.upbound.io_channels.yaml`
Manually:
- `package/crds/networkmanager.aws.upbound.io_corenetworks.yaml`
Uptest:
- `package/crds/rds.aws.upbound.io_instances.yaml`
Manually:
- `package/crds/redshift.aws.upbound.io_clusters.yaml`
Uptest:
- `package/crds/secretsmanager.aws.upbound.io_secrets.yaml`
Uptest:
- `package/crds/transfer.aws.upbound.io_servers.yaml`
Uptest:


**Native schema version changes**

- `aws_autoscaling_group`
- `aws_db_instance`
- `aws_dx_connection`
- `aws_lambda_provisioned_concurrency_config`

**Comment by turkenf** [comment]
/test-examples="examples/connect/hoursofoperation.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/directconnect/connection.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/defaultvpc.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/connect/queue.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/route.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/connect/routingprofile.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/routetable.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/routetableassociation.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpcpeeringconnectionaccepter.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/elasticache/cluster.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/iam/role.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpcpeeringconnectionoption.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/elasticache/replicationgroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/lightsail/instance.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/networkmanager/corenetwork.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/redshift/cluster.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/secretsmanager/secret.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/transfer/server.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpcpeeringconnectionoptions.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpcpeeringconnection.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/elasticache/replicationgroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/firehose/deliverystream.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/rds/instance.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/lambda/provisionedconcurrencyconfig.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/ec2/vpc.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/elasticache/replicationgroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/rds/instance.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/kafka/cluster.yaml"

**Comment by turkenf** [comment]
When I test resource `Instance.rds` with the current configuration and example in this PR, I get the following error:
```
    message: 'observe failed: cannot run refresh: refresh failed: only lowercase alphanumeric
      characters and hyphens allowed in "identifier": '
```
This issue seems to be caused by [this change](https://github.com/hashicorp/terraform-provider-aws/pull/31232) on the terraform side.

I did not get any successful results in my tests with the current external name configuration `config.ParameterAsIdentifier("identifier")`. When I use the `config.IdentifierFromProvider` configuration, since the `identifier` parameter comes `spec.forProvider`, there is a conflict with the `identifier_prefix` parameter. When I moved the `identifier_prefix` parameter to `IgnoredFields`, the resource could be created successfully.
```
NAME                                       READY   SYNCED   EXTERNAL-NAME                          AGE
key.kms.aws.upbound.io/sample-key-tftest   True    True     d5f6f817-5a3a-48e4-a8f9-424cb87ba7c7   88m

NAME                                                    READY   SYNCED   EXTERNAL-NAME                   AGE
instance.rds.aws.upbound.io/example-dbinstance-tftest   True    True     db-DVG3KOZWALZJBAQE5CSQULB3QU   88m
```

Here I need confirmation to make this change, I'm not sure this issue can be solved without changing the external name configuration.

**Comment by turkenf** [comment]
/test-examples="examples/autoscaling/autoscalinggroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/autoscaling/launchconfiguration.yaml"

**Comment by turkenf** [comment]
While testing the `Cluster.kafka` resource, there were deprecated parameters in the connected resources. After fixing this resource and its dependencies, I got the following error:
```
    message: |-
      observe failed: cannot run refresh: refresh failed: Missing required argument: The argument "arn" is required, but no definition was found.
      Missing required argument: The argument "revision" is required, but no definition was found.
```
Although the `configurationInfo` parameter is not required, we need to pass the `arn` and `revision` (required for the configurationInfo block) parameters because it comes as an empty set.
```
    configurationInfo:
    - {}
```
So I added the `Configuration.kafka` resource to the `Cluster.kafka` example, manually filled in the `arn` and `revision` parameters and added a manual intervention.

Here, this solution can be applied as a temporary solution, but it may need to be investigated further. Since it takes more than 120m to create and delete the `Cluster.kafka` resource, I leave it this way for now.

**Comment by mbbush** [comment]
I think you ran into https://github.com/upbound/provider-aws/issues/489

**Comment by mbbush** [comment]
@turkenf Do you think something like this would work to make a selector for the Configuration?

Also, I'd love it if you could bump to native provider 5.16.2, which contains a bugfix I contributed which solves #748

File: config/provider-metadata.yaml [hunk 1]
```diff
@@ -11039,7 +11399,7 @@ resources:
         importStatements: []
     aws_cloudfront_cache_policy:
         subCategory: CloudFront
-        description: Provides a cache policy for a CloudFront ditribution. When it’s attached to a cache behavior, the cache policy determines the values that CloudFront includes in the cache key. These values can include HTTP headers, cookies, and URL query strings. CloudFront uses the cache key to find an object in its cache that it can return to the viewer. It also determines the default, minimum, and maximum time to live (TTL) values that you want objects to stay in the CloudFront cache.
+        description: Use the
```
Comment (mbbush) [inline config/provider-metadata.yaml:11402]: This doesn't seem right. Maybe the parser in the code generator ran into something it couldn't handle?

File: config/provider-metadata.yaml [hunk 2]
```diff
@@ -15885,7 +16462,7 @@ resources:
         importStatements: []
     aws_cognito_managed_user_pool_client:
         subCategory: Cognito IDP (Identity Provider)
-        description: Manages a Cognito User Pool Client resource created by another service.
+        description: Use the
```
Comment (mbbush) [inline config/provider-metadata.yaml:16465]: This looks like it got truncated too

**Comment by turkenf** [comment]
@mbbush, thank you for your review and contributions 🙏 

> @turkenf Do you think something like [this](https://github.com/upbound/provider-aws/commit/72a588997f80c8218f63f87f05765cab414a7578) would work to make a selector for the Configuration? I haven't tested it, but I'd be happy to do so if you think it's a worthwhile approach to try.

I need to do more investigation and talk to the team to find the correct solution to the error received for the `Cluster.kafka` resource. Maybe we can also consider moving the `configurationInfo` block to `IgnoreFields`. But in both cases, there is no harm in adding selectors for these two parameters, regardless of this PR.

> Also, I'd love it if you could bump to native provider 5.16.2, which contains a https://github.com/hashicorp/terraform-provider-aws/pull/33380 I contributed which solves https://github.com/upbound/provider-aws/issues/748

I will evaluate this and check if we can upgrade to version 5.16.2

**Comment by mbbush** [comment]
I'd love to be able to have a single selector that extracts both the arn and revision from the status of the Configuration, but I can't see how to do that within the current structure. I guess `configurationInfo.arnSelector` and `configurtationInfo.revisionSelector` is still an improvement over the current state.

**Review by mbbush — COMMENTED** [review]
There was a third description that got truncated as well.

**Comment by mbbush** [comment]
@turkenf do you think this provider upgrade is going to make it into this month's release, or likely slip to next. Is there anything I can do to help it along?

**Comment by turkenf** [comment]
> @turkenf do you think this provider upgrade is going to make it into this month's release, or likely slip to next? Is there anything I can do to help it along?

@mbbush, thank you for your interest.
We will upgrade the native provider version to v5.18.1, seems that this PR will be included in the next release.

**Comment by sergenyalcin** [comment]
/test-examples="examples/ec2/vpc.yaml"

**Comment by ulucinar** [comment]
Hi @turkenf,
While rebasing the PR, I resolved a conflict in `examples/kafka/cluster.yaml` by accepting the @mbbush's changes. Changes from the feature branch were adding the `manual-intervention` annotation whereas Matthew has added the `Configuration.kafka` resource as a dependency. I assume Matthew's changes make the resource uptestable. Will trigger an uptest below.

**Comment by ulucinar** [comment]
/test-examples="examples/kafka/cluster.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/kafka/cluster.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/vpc.yaml"

**Comment by turkenf** [comment]
After bumping version 5.15.0 to 5.19.0:

Breaking CRD schema changes

- `package/crds/sesv2.aws.upbound.io_emailidentities.yaml`

Native schema version changes

- `aws_servicecatalog_principal_portfolio_association`

**Comment by turkenf** [comment]
/test-examples="examples/sesv2/emailidentity.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/rds/instance.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/sesv2/emailidentity.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/servicecatalog/principalportfolioassociation.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/docdb/cluster.yaml"

**Comment by turkenf** [comment]
test-examples="examples/docdb/clusterparametergroup.yaml"

**Comment by turkenf** [comment]
test-examples="examples/docdb/clusterparametergroup.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/docdb/clustersnapshot.yaml"

**Comment by turkenf** [comment]
Covered in https://github.com/upbound/provider-aws/pull/1037

**Pull request closed without merging**